src/modules/adaptors/Frax/CollateralFTokenAdaptorV2.sol

```diff
@@ -0,0 +1,261 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity 0.8.21;
+
+import { BaseAdaptor, ERC20, SafeTransferLib, Cellar, PriceRouter, Math } from "src/modules/adaptors/BaseAdaptor.sol";
+import { IFToken } from "src/interfaces/external/Frax/IFToken.sol";
+import { FraxlendHealthFactorLogic } from "src/modules/adaptors/Frax/FraxlendHealthFactorLogic.sol";
+
+/**
+ * @title FraxLend Collateral Adaptor
+ * @notice Allows addition and removal of collateralAssets to Fraxlend pairs for a Cellar.
+ * @author crispymangoes, 0xEinCodes
+ */
+contract CollateralFTokenAdaptorV2 is BaseAdaptor, FraxlendHealthFactorLogic {
+    using SafeTransferLib for ERC20;
+    using Math for uint256;
+
+    //==================== Adaptor Data Specification ====================
+    // adaptorData = abi.encode(IFToken fraxlendPair)
+    // Where:
+    // `fraxlendPair` is the fraxlend pair this adaptor position is working with. It is also synomous to fToken used in `FTokenAdaptor.sol` and `FTokenAdaptorV1.sol`
+    //================= Configuration Data Specification =================
+    // N/A because the DebtFTokenAdaptor handles actual deposits and withdrawals.
+    // ==================================================================
+
+    /**
+     * @notice Attempted to interact with an fraxlendPair the Cellar is not using.
+     */
+    error CollateralFTokenAdaptor__FraxlendPairPositionsMustBeTracked(address fraxlendPair);
+
+    /**
+     * @notice Removal of collateral causes Cellar Health Factor below what is required
+     */
+    error CollateralFTokenAdaptor__HealthFactorTooLow(address fraxlendPair);
+
+    /**
+     * @notice The FRAX contract on current network.
+     * @notice For mainnet use 0x853d955aCEf822Db058eb8505911ED77F175b99e.
+     */
+    ERC20 public immutable FRAX;
+
+    /**
+     * @notice Minimum Health Factor enforced after every removeCollateral() strategist function call.
+     * @notice Overwrites strategist set minimums if they are lower.
+     */
+    uint256 public immutable minimumHealthFactor;
+
+    constructor(address _frax, uint256 _healthFactor) {
+        _verifyConstructorMinimumHealthFactor(_healthFactor);
+        FRAX = ERC20(_frax);
+        minimumHealthFactor = _healthFactor;
+    }
+
+    //============================================ Global Functions ===========================================
+    /**
+     * @dev Identifier unique to this adaptor for a shared registry.
+     * Normally the identifier would just be the address of this contract, but this
+     * Identifier is needed during Cellar Delegate Call Operations, so getting the address
+     * of the adaptor is more difficult.
+     */
+    function identifier() public pure virtual override returns (bytes32) {
+        return keccak256(abi.encode("FraxLend Collateral fToken Adaptor V 0.1"));
```

Comment:
Change name to ...fTokenV2 Adaptor... to be more consistent with V1 Collateral adaptor?

Reply:
Done